mmengine/runner/runner.py

```diff
@@ -236,13 +239,14 @@ class Runner:
     def __init__(
         self,
         model: Union[nn.Module, Dict],
-        work_dir: str,
+        work_dir: Optional[str] = None,
         train_dataloader: Optional[Union[DataLoader, Dict]] = None,
         val_dataloader: Optional[Union[DataLoader, Dict]] = None,
         test_dataloader: Optional[Union[DataLoader, Dict]] = None,
         train_cfg: Optional[Dict] = None,
         val_cfg: Optional[Dict] = None,
         test_cfg: Optional[Dict] = None,
+        auto_distinguish_experiment: Optional[bool] = True,
```

Comment:
`auto_distinguish_experiment` could be hidden into the `cfg`.

Reply:
```suggestion
        auto_distinguish_experiment: bool = True,
```
`Optional` is not necessary (Optional is only for argument with default value `None`).

Reply:
> `auto_distinguish_experiment` could be hidden into the `cfg`.

Do you mean that we should delete the `auto_distinguish_experiment` param from the runner init parameters, and get it by cfg.get('auto_distinguish_experiment', True)?

Reply:
> `Optional` is not necessary (Optional is only for argument with default value `None`).

Got it.